NuKeeper.Gitlab/GitlabPlatform.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Threading.Tasks;
+using NuKeeper.Abstractions.CollaborationModels;
+using NuKeeper.Abstractions.CollaborationPlatform;
+using NuKeeper.Abstractions.Configuration;
+using NuKeeper.Abstractions.Logging;
+using NuKeeper.Gitlab.Model;
+using User = NuKeeper.Abstractions.CollaborationModels.User;
+
+namespace NuKeeper.Gitlab
+{
+    public class GitlabPlatform : ICollaborationPlatform
+    {
+        private readonly INuKeeperLogger _logger;
+        private GitlabRestClient _client;
+        private AuthSettings _settings;
+
+        public GitlabPlatform(INuKeeperLogger logger)
+        {
+            _logger = logger;
+        }
+
+        public void Initialise(AuthSettings settings)
+        {
+            _settings = settings;
+
+            var httpClient = new HttpClient
+            {
+                BaseAddress = settings.ApiBase
+            };
+            _client = new GitlabRestClient(httpClient, settings.Token, _logger);
+        }
+
+        public async Task<User> GetCurrentUser()
+        {
+            var user = await _client.GetCurrentUser().ConfigureAwait(false);
+
+            return new User(user.UserName, user.Name, user.Email);
+        }
+
+        public async Task OpenPullRequest(ForkData target, PullRequestRequest request, IEnumerable<string> labels)
+        {
+            var projectName = target.Owner;
+            var repositoryName = target.Name;
+
+            var mergeRequest = new MergeRequest
+            {
+                Title = request.Title,
+                SourceBranch = request.Head,
+                Description = request.Body,
+                TargetBranch = request.BaseRef,
+                Id = $"{projectName}/{repositoryName}"
+            };
+
+            await _client.OpenMergeRequest(projectName, repositoryName, mergeRequest).ConfigureAwait(false);
```

Comment:
I do agree here, however gitlab is actually calling it Merge Request instead of Pull Request, so it might be cleaner to keep it this way here as it's anyway a platfrom specific code.

Reply:
Obviously your call :)

Reply:
So `Azure DevOps` uses `CreatePullRequest` and Github uses `OpenPullRequest`..

Reply:
Keeping platform standards is also fine